corehq/apps/locations/models.py

```diff
@@ -761,3 +766,19 @@ def _unassign_users_from_location(domain, location_id):
             user.unset_location_by_id(domain, location_id, fall_back_to_next=True)
         elif user.is_commcare_user():
             user.unset_location_by_id(location_id, fall_back_to_next=True)
+
+
+def notify_users_at_location(location_id):
```

Comment:
I know you didn't start it, but is notify the right word here? I might also just not be understanding what this does but from a quick read of the code, I would have said it's _updating_ users at this location. Or maybe "propagating location change to users". Will let you and/or the original authors of this code decide whether changing that name is something you want to get into.

Reply:
this also might be best done in a celery task since it could end up iterating through a lot of users

Reply:
Also, if you look at the implementation of `update_fixture_status`, it doesn't require anything of the user class beyond `._id`.  (You should be able to invalidate the `quickcache` from the class, without an instance).  Pulling and wrapping users can actually take quite a while, so I'd make that method available as a standalone function and skip the db calls for users.  Actually, it might be best to make an analogous function which accepts a list of user_ids and operates in chunks using `queryset.update()`.  Something like:
```python
for ids in chunked(user_ids):
    (UserFixtureStatus.objects
     .filter(user_id__in=ids,
             fixture_type=fixture_type)
     .update(last_modified=now))
```